src/mem/alloc.h

```diff
@@ -860,56 +848,39 @@ namespace snmalloc
       {
         error("Critical error: Out-of-memory during initialisation.");
       }
-      dummy->set_target_id(id());
+      dummy->set_info(get_trunc_id(), size_to_sizeclass_const(MIN_ALLOC_SIZE));
       message_queue().init(dummy);
     }
 
     SNMALLOC_FAST_PATH void handle_dealloc_remote(Remote* p)
     {
-      Superslab* super = Superslab::get(p);
+      if (likely(p->trunc_target_id() == get_trunc_id()))
+      {
+        // Destined for my slabs
+        Superslab* super = Superslab::get(p);
 
 #ifdef CHECK_CLIENT
-      if (p->target_id() != super->get_allocator()->id())
-        error("Detected memory corruption.  Potential use-after-free");
+        if (p->trunc_target_id() != (super->get_allocator()->trunc_id()))
+          error("Detected memory corruption.  Potential use-after-free");
 #endif
-      if (likely(super->get_kind() == Super))
-      {
-        Slab* slab = Metaslab::get_slab(p);
-        Metaslab& meta = super->get_meta(slab);
-        if (likely(p->target_id() == id()))
-        {
-          small_dealloc_offseted(super, p, meta.sizeclass);
-          return;
-        }
-      }
-      handle_dealloc_remote_slow(p);
-    }
+        // Guard against remote queues that have colliding IDs
+        SNMALLOC_ASSERT(likely(super->get_allocator() == public_state()));
 
-    SNMALLOC_SLOW_PATH void handle_dealloc_remote_slow(Remote* p)
-    {
-      Superslab* super = Superslab::get(p);
-      if (likely(super->get_kind() == Medium))
-      {
-        Mediumslab* slab = Mediumslab::get(p);
-        if (p->target_id() == id())
+        if (likely(super->get_kind() == Super))
```

Comment:
Can't you use the sizeclass here to check which path. 
```suggestion
if (p->sizeclass() < NUM_SMALL_SIZECLASSES)
```

Reply:
Sure, that also works.  One fewer load?